app/models/agency.rb

```diff
@@ -71,4 +71,20 @@ def slug_candidates
   def retrieve_state
     State.where(id: state_id).pluck(:name).join
   end
+
+  def location
+    @location ||= Location.new(
+      state: state.name,
+      street_location: street_address,
+      city: city,
+      zipcode: zipcode
+    )
+  end
+
+  def location=(location)
```

Comment:
Hmm, I'm not quite sure why you had to add this?

Reply:
I was following the example from the issue [here](https://revs.runtime-revolution.com/value-objects-in-ruby-on-rails-9df64bc8db34) to create a setter method for `Location` value object. I think that's why the examples were non-memorized 🤔 , otherwise if it was memoized, then I think we would get the original `Location` object even if we updated an `Agency`'s address during runtime.

Reply:
Oh, ok - in that case, it might be better not to memoize it.  Sorry, but do you mind removing it?  This is really is the last thing, haha.

Reply:
Not a problem at all 🙂

Reply:
Done! Thank you for responding so quickly.